test/integration/apiserver/apply/reset_fields_test.go

```diff
@@ -85,60 +92,106 @@ func TestApplyResetFields(t *testing.T) {
 				t.Fatal(err)
 			}
 			t.Run(mapping.Resource.String(), func(t *testing.T) {
-				status, ok := statusData[mapping.Resource]
-				if !ok {
-					status = statusDefault
+				// TODO: delete this after updating test data for all resources
+				if mapping.Resource.Resource != "services" {
+					t.Skip()
 				}
-
-				newResource, ok := createData[mapping.Resource]
-				if !ok {
-					t.Fatalf("no test data for %s.  Please add a test for your new type to etcd.GetEtcdStorageData().", mapping.Resource)
-				}
-
-				newObj := unstructured.Unstructured{}
-				if err := json.Unmarshal([]byte(newResource.Stub), &newObj.Object); err != nil {
-					t.Fatal(err)
-				}
-				if err := json.Unmarshal([]byte(status), &newObj.Object); err != nil {
-					t.Fatal(err)
-				}
-
 				namespace := resetFieldsNamespace
 				if mapping.Scope == meta.RESTScopeRoot {
 					namespace = ""
 				}
-				name := newObj.GetName()
-				rsc := dynamicClient.Resource(mapping.Resource).Namespace(namespace)
-				_, err := rsc.Create(context.TODO(), &newObj, metav1.CreateOptions{FieldManager: "create_test"})
+
+				// create first object
+				obj1 := createObject(t, mapping, statusDefault, statusData, createData1)
+				obj1YAML, err := yaml.Marshal(obj1.Object)
 				if err != nil {
 					t.Fatal(err)
 				}
+				name := obj1.GetName()
 
-				statusObj := unstructured.Unstructured{}
-				if err := json.Unmarshal([]byte(status), &statusObj.Object); err != nil {
-					t.Fatal(err)
+				// apply the spec of the first object
+				apply1, err := dynamicClient.
+					Resource(mapping.Resource).
+					Namespace(namespace).
+					Patch(context.TODO(), obj1.GetName(), types.ApplyPatchType, obj1YAML, metav1.PatchOptions{FieldManager: "fieldmanager1"}, "")
+				if err != nil {
+					t.Fatalf("Failed to apply obj1: %v", err)
 				}
-				statusObj.SetAPIVersion(mapping.GroupVersionKind.GroupVersion().String())
-				statusObj.SetKind(mapping.GroupVersionKind.Kind)
-				statusObj.SetName(name)
-				statusYAML, err := yaml.Marshal(statusObj.Object)
+				fmt.Printf("apply1 = %+v\n", apply1)
+
+				// create second object
+				obj2 := createObject(t, mapping, resetFieldsStatusDefault, resetFieldsStatusData, createData2)
```

Comment:
I don't understand the second create?

Reply:
oh this doesn't actually create the object, I see. Maybe call it "assembleObject" or something?

Reply:
maybe create is not the right term. I'm constructing the yaml to be applied (with a different spec and status than the first object). Would it be more clear if it was called `constructObject` or am I missing a simple way of injecting the modified status and spec into the initial object. 

Are you suggesting I `json.Unmarshal` the new spec and status directly into the existing object? I guess I could do that and pass an object into `createObject()` (renamed to `injectSpecAndStatus()`  or something). Is that what you mean?

Reply:
oops, I keep commenting without refreshing the page and not seeing your newer comments. Sure I can rename to `assembleObject`

Reply:
The code is fine, just the name "create" made me think that was making a POST call.